### PR TITLE
Feat/config file

### PR DESCRIPTION
**Pull request by Mopsgeschwindigkeit**
# Summary - *The What* #
**DO** - I have added the possibility to set configuration inside a config file. Either by storing it in  `/etc/sxapi.conf`, `~/.config/sxapi.confg` or by providing an alternative path during the cli command with the `-c|--configfile` flag.

The config priorities are Cli-Parameter > Envirionmet Vars > ConfigFile.

I also renamed the `UserCredentials` class to `CliUser` and added the api-clients as attributes. I did this to provide a  place where the cli-command can get already instantiated api-clients.

# Magic - *The How* (Optional)
**DO** - For parsing the config files i used the stdlib `configparser`.

# ToDo's - *Open Point List* #
**DO**
* Using the configs `ORGA` field inside the commands/calls.
* Writing Documentation

### CODE REVIEW

File: src/sxapi/cli/cli_user.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    """
+
+    def __init__(self):
+        """
+        Basic User Credentials Constructor
+
+        calls self._init_creds() to set available credentials on startup.
+        """
+
+        self.token_env_names = "SMAXTEC_TOKEN"
```
Comment (flowolf) [inline src/sxapi/cli/cli_user.py:27]: please specify what kind of token this is. 

we do have several token types in our system. Access token api, firestore tokens, long lived access tokens, etc 

we might also have different tokens for read/write or certain APIs.

`self.api_access_token` might suffice

File: src/sxapi/cli/example-config.conf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[SXAPI]
+
+# smaXtec user name
+USER = smaxtec@user.com
```
Comment (flowolf) [inline src/sxapi/cli/example-config.conf:4]: please use example.com as domain for this:

```suggestion
USER = user.name@example.com
```

File: src/sxapi/cli/example-config.conf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+# smaXtec password
+PASSWORD = smaxtec_user_pwd
```
Comment (flowolf) [inline src/sxapi/cli/example-config.conf:7]:
```suggestion
PASSWORD = "userSecret"
```

File: tests/cli_tests/test-config.conf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[SXAPI]
+
+USER = test@user.com
```
Comment (flowolf) [inline tests/cli_tests/test-config.conf:3]:
```suggestion
USER = test@example.com
```

File: tests/cli_tests/test-config_param.conf [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[SXAPI]
+
+USER = test2@user.com
```
Comment (flowolf) [inline tests/cli_tests/test-config_param.conf:3]:
```suggestion
USER = test2@example.com
```

File: tests/cli_tests/test_cli.py [hunk 6]
```diff
@@ -29,8 +30,95 @@ def test_func(keyring_mock, print_mock, version_mock):
...
+@mock.patch("sxapi.cli.cli_user.get_token_keyring", return_value="keyring-token")
+def test_config(keyring_mock, version_mock):
+    # test config file default
+    cli.config_file_paths = ["./tests/cli_tests/test-config.conf"]
+    with mock.patch("sys.argv", ["sx_api", "--version"]):
+        with mock.patch("sxapi.cli.cli_user.init_user") as iu_mock:
+            cli.run()
+            res = iu_mock.call_args_list[0][0][0]
+
+            assert res["user"] == "test@user.com"
```
Comment (flowolf) [inline tests/cli_tests/test_cli.py:50]:
```suggestion
            assert res["user"] == "test@example.com"
```

File: tests/cli_tests/test_cli.py [hunk 7]
```diff
@@ -29,8 +30,95 @@ def test_func(keyring_mock, print_mock, version_mock):
...
+    # test config_file_path override
+    with mock.patch(
+        "sys.argv",
+        ["sx_api", "--version", "-c", "./tests/cli_tests/test-config_param.conf"],
+    ):
+        with mock.patch("sxapi.cli.cli_user.init_user") as iu_mock:
+            cli.run()
+            res = iu_mock.call_args_list[0][0][0]
+
+            assert res["user"] == "test2@user.com"
```
Comment (flowolf) [inline tests/cli_tests/test_cli.py:65]:
```suggestion
            assert res["user"] == "test2@example.com"
```

File: tests/cli_tests/test_cli.py [hunk 8]
```diff
@@ -29,8 +30,95 @@ def test_func(keyring_mock, print_mock, version_mock):
...
+            assert res["pwd"] == "smaxtec_test2_user_pwd"
+            assert res["orga"] == "smaxtec_test2_organisation_id"
+            assert res["api_public_v2_path"] == "test2_path_test"
+            assert res["api_integration_v2_path"] == "test2_path_integration"
+
+    # test override with env_vars
+    os.environ["SXAPI_USER"] = "testenv@user.com"
```
Comment (flowolf) [inline tests/cli_tests/test_cli.py:72]:
```suggestion
    os.environ["SXAPI_USER"] = "testenv@example.com"
```

File: tests/cli_tests/test_cli.py [hunk 9]
```diff
@@ -29,8 +30,95 @@ def test_func(keyring_mock, print_mock, version_mock):
...
+    with mock.patch(
+        "sys.argv", ["sx_api", "--version", "-c", "./test-config_param.conf"]
+    ):
+        with mock.patch("sxapi.cli.cli_user.init_user") as iu_mock:
+            cli.run()
+            res = iu_mock.call_args_list[0][0][0]
+
+            assert res["user"] == "testenv@user.com"
```
Comment (flowolf) [inline tests/cli_tests/test_cli.py:80]:
```suggestion
            assert res["user"] == "testenv@example.com"
```

**Review by flowolf — COMMENTED** [review]
looking good. just a few small adjustments.

File: src/sxapi/cli/cli_user.py (hunk 1 above)
Comment (Mopsgeschwindigkeit) [reply src/sxapi/cli/cli_user.py]: renamed to `self.api_access_token` :)

**Pull request merged**